src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/ResourceTypeIdParameterQueryGenerator.cs

```diff
@@ -22,5 +24,24 @@ public override SearchParameterQueryGeneratorContext VisitString(StringExpressio
 
             return VisitSimpleBinary(BinaryOperator.Equal, context, VLatest.Resource.ResourceTypeId, expression.ComponentIndex, resourceTypeId);
         }
+
+        public override SearchParameterQueryGeneratorContext VisitIn(InExpression expression, SearchParameterQueryGeneratorContext context)
+        {
+            List<object> resolvedResourceTypeIds = new List<object>(capacity: expression.Values.Count);
+
+            foreach (string resourceType in expression.Values)
+            {
+                if (context.Model.TryGetResourceTypeId(resourceType, out short resourceTypeId))
+                {
+                    resolvedResourceTypeIds.Add(resourceTypeId);
+                }
+                else
+                {
+                    throw new InvalidOperationException($"Invalid resource type '{resourceType}'.");
```

Comment:
I know we shouldn't end up here, but worth asking if we already have a resource string that covers this text